components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/PAPStatusDataHandlerTest.java

```diff
@@ -0,0 +1,330 @@
+import static org.wso2.carbon.identity.entitlement.common.EntitlementConstants.StatusTypes.PUBLISH_POLICY;
+import static org.wso2.carbon.identity.entitlement.persistence.SubscriberPersistenceManagerTest.SUBSCRIBER_MODULE_NAME;
+
+/**
+ * This is the parent test class for the PAP Status Data Handler test classes.
+ */
+public abstract class PAPStatusDataHandlerTest {
+
+    static final String SUBSCRIBER_ID_KEY = "subscriberId";
+    static final String SUBSCRIBER_ID_DISPLAY_NAME = "Subscriber Id";
+
+    protected static final String ABOUT_POLICY = "POLICY";
+    protected static final String ABOUT_SUBSCRIBER = "SUBSCRIBER";
+    protected static final String POLICY_KEY = "simple_policy";
+    protected static final String SUBSCRIBER_KEY = "PDP_Subscriber_test";
+    protected static final String PAP_POLICY_STORE_TARGET = "PAP POLICY STORE";
+    protected static final String TARGET_ACTION_PERSIST = "PERSIST";
+    protected static final String TARGET_ACTION_REMOVE = "REMOVE";
+    protected static final String TARGET_ACTION_LOAD = "LOAD";
+    protected static final String POLICY_VERSION_1 = "1";
+    protected static final String SAMPLE_USER = "admin";
+
+    StatusHolder statusHolderForAddPolicy;
+    StatusHolder statusHolderForGetPolicy;
+    StatusHolder statusHolderForRemovePolicy;
+
+    StatusHolder statusHolderForAddPolicyInPDP;
+    StatusHolder statusHolderForUpdatePolicyInPDP;
+    StatusHolder statusHolderForRemovePolicyInPDP;
+
+    List<StatusHolder> policyStatusHolderList1;
+    List<StatusHolder> policyStatusHolderList2;
+    List<StatusHolder> policyStatusHolderList3;
+
+    List<StatusHolder> pdpStatusHolderList1;
+    List<StatusHolder> pdpStatusHolderList2;
+    List<StatusHolder> pdpStatusHolderList3;
+
+    PublisherDataHolder publisherDataHolder;
+
+    PAPStatusDataHandler papStatusDataHandler;
+    SubscriberPersistenceManager subscriberPersistenceManager;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        Properties engineProperties = new Properties();
+        engineProperties.put(PDPConstants.MAX_NO_OF_STATUS_RECORDS, "5");
+        EntitlementConfigHolder.getInstance().setEngineProperties(engineProperties);
+
+        Properties storeProps = new Properties();
+        storeProps.put(PDPConstants.MAX_NO_OF_STATUS_RECORDS, "5");
+        papStatusDataHandler = createPAPStatusDataHandler();
+        papStatusDataHandler.init(storeProps);
+
+        subscriberPersistenceManager = createSubscriberPersistenceManager();
+        setupPolicyStatusData();
+        setupSubscriberStatusData();
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+
+        papStatusDataHandler.handle(ABOUT_POLICY, POLICY_KEY, policyStatusHolderList3);
+        papStatusDataHandler.handle(ABOUT_SUBSCRIBER, SUBSCRIBER_KEY, pdpStatusHolderList3);
+        subscriberPersistenceManager.removeSubscriber(SUBSCRIBER_KEY);
```

Comment:
what exactly we do here? is it about resetting data to a base level?

if so, its bit odd to use the methods that we want to test, to do the resetting.. If there is an issue in the `handle()` method or `remove()` method, many other tests will also fails.. making the identifying actual root cause for such failure will be harder due to the unnecessary noise..

Reply:
Yes. This is about resetting data for empty state. There is no any other method to remove PAP Status related data. This status data removing part also done from the `handle()` method based on the Status type `DELETE_POLICY`.